src/utils.ts

```diff
@@ -1314,36 +1314,30 @@ export function isHostMatch(match: RegExp, host?: string): boolean {
  */
 export async function abortable<T>(
   promise: Promise<T>,
-  { signal }: { signal?: AbortSignal } = {}
+  { signal }: { signal: AbortSignal }
 ): Promise<T> {
-  const { abort, done } = aborted(signal);
+  const { promise: aborted, reject } = promiseWithResolvers<never>();
+
+  function rejectOnAbort() {
+    reject(signal.reason);
+  }
+
+  if (signal.aborted) rejectOnAbort();
```

Comment:
It is not necessary, we attach rejection handling synchronously because there is no interruption between this point and the line where we call `Promise.race`. We have test cases demonstrating what an already rejected abort signal does with this API. 

We want to create a rejected promise but we still want to race with the input promise so that if it is rejected the error from that promise takes priority. This is also why we can't use throwIfAborted here, it would make the signal's error come before the input promise's one.